### PR TITLE
Set `NODE_ENV` when generating bundler

### PR DESCRIPTION
**Pull request by williamchong**
Unless there are some other implications I missed, this simple line should solve https://github.com/MetaMask/metamask-extension/issues/4115

### CODE REVIEW

**Comment by bdresser** [comment]
@danjm would you take a look when you have a chance?

File: gulpfile.js [hunk 1]
```diff
@@ -449,10 +449,12 @@ function generateBundler (opts, performBundle) {
 
   let bundler = browserify(browserifyOpts)
 
+  process.env.NODE_ENV = opts.devMode ? 'development' : 'production'
```
Comment (kumavis) [inline gulpfile.js]: @williamchong007 do we need to set `process.env.NODE_ENV` or just in the bundle?

**Comment by williamchong007** [comment]
I doubled checked and found a (seemingly) better approach to this @kumavis 
the problem is envify `NODE_ENV` did not affect `react` in `node_modules`

https://github.com/hughsk/envify/issues/56#issuecomment-302384436

**Review by whymarrh — DISMISSED** [review]
I've tested this locally and it seems to work as expected (also helps a small bit with our bundle size)

**Comment by whymarrh** [comment]
I rebased this on `develop`

**Review by whymarrh — APPROVED** [review]
LGTM, thanks @williamchong007!

**Pull request merged**